v2/apiserver/internal/core/mongodb/events_store.go

```diff
@@ -71,16 +71,27 @@ func (e *eventsStore) List(
 	events := core.EventList{}
 
 	criteria := bson.M{
-		"worker.status.phase": bson.M{
-			"$in": selector.WorkerPhases,
-		},
 		"deleted": bson.M{
 			"$exists": false, // Don't grab logically deleted events
 		},
 	}
 	if selector.ProjectID != "" {
 		criteria["projectID"] = selector.ProjectID
 	}
+	if selector.Source != "" {
```

Comment:
Should we consider consolidating the shared bits of criteria-building based on the provided selector, in a similar vein to `eventsSelectorToQueryParams`, etc., added in this PR?

Reply:
We could, but I had opted against it. Reasons being:

1. For `CancelMany` and `DeleteMany`, `ProjectID` is required (and the service layer enforces that), but for `List`, it's optional, so it comes with an extra `if selector.ProjectID != ""...` guard that doesn't occur in the other two places.

2. For `CancelMany` the criteria based on worker phase are different. We build and execute two separate statements for things that need to go from phase `PENDING` to `CANCELED` and things that need to go from `STARTING` or `RUNNING` to `ABORTED`.

In light of these differences, we could use a common function to handle extracting _some_ query/statement criteria from the selector, but for other bits of criteria, we'd need varying, more specific logic. It felt to me like it would be more confusing to spread the criteria-building logic around than to just keep it how it is here, even if that means being a little bit duplicative.

All that said, I am, as ever, willing to entertain any counter-argument to that.

Reply:
Appreciate your pointing out the finer-grained differences between these functions.  Let's leave as is for now!